src/test/integration/tests/Deposit_Delegate_Queue_Complete.t.sol

```diff
@@ -10,6 +10,70 @@ contract Deposit_Delegate_Queue_Complete is IntegrationBase {
                                 FULL WITHDRAWALS
     *******************************************************************************/
 
+    function testFuzz_deposit_delegate_updateBalance(uint24 _random) public {
+        _configRand({
+            _randomSeed: _random,
+            _assetTypes: HOLDS_LST | HOLDS_ETH | HOLDS_ALL,
+            _userTypes: DEFAULT | ALT_METHODS
+        });
+
+        (
+            User staker,
+            IStrategy[] memory strategies, 
+            uint[] memory tokenBalances
+        ) = _newRandomStaker();
+        (User operator, ,) = _newRandomOperator();
+        uint[] memory shares = _calculateExpectedShares(strategies, tokenBalances);
+
+        assert_HasNoDelegatableShares(staker, "staker should not have delegatable shares before depositing");
+        assertFalse(delegationManager.isDelegated(address(staker)), "staker should not be delegated");
+
+        staker.depositIntoEigenlayer(strategies, tokenBalances);
+
+        assert_HasNoUnderlyingTokenBalance(staker, strategies, "staker should have transferred all underlying tokens");
+        assert_Snap_Added_StakerShares(staker, strategies, shares, "staker should expected shares in each strategy after depositing");
+
+        /// 2. Delegate to an operator:
+        //
+        // ... check that the staker is now delegated to the operator, and that the operator
+        //     was awarded the staker's shares
+        staker.delegateTo(operator);
+
+        assertTrue(delegationManager.isDelegated(address(staker)), "staker should be delegated");
+        assertEq(address(operator), delegationManager.delegatedTo(address(staker)), "staker should be delegated to operator");
+        assert_Snap_Unchanged_StakerShares(staker, "staker shares should be unchanged after delegating");
+        assert_Snap_Added_OperatorShares(operator, strategies, shares, "operator should have received shares");
+
+        IDelegationManager.Withdrawal[] memory withdrawals;
+        bytes32[] memory withdrawalRoots;
+        /// 3. Queue withdrawal(s):
+        // The staker will queue one or more withdrawals for all strategies and shares
+        //
+        // ... check that each withdrawal was successfully enqueued, that the returned withdrawals
+        //     match now-pending withdrawal roots, and that the staker and operator have
+        //     reduced shares.
+        withdrawals = staker.queueWithdrawals(strategies, shares);
+        withdrawalRoots = _getWithdrawalHashes(withdrawals);
+
+        assert_ValidWithdrawalHashes(withdrawals, withdrawalRoots, "calculated withdrawals should match returned roots");
+        assert_AllWithdrawalsPending(withdrawalRoots, "staker's withdrawals should now be pending");
```

Comment:
Nit: I've been trying to not use apostrophes since the highlighting below gets messed up